scripts/validate-autoinstall-user-data.py

```diff
@@ -29,56 +29,117 @@
 
 import argparse
 import io
+from textwrap import dedent
+from typing import Callable
 import json
 
 import jsonschema
 import yaml
 
 
+def parse_args() -> argparse.Namespace:
+    """Parse arguments with argparse"""
+
+    description: str = dedent(
+        """\
+    Validate autoinstall user data against the autoinstall schema. By default
+    expects the user data is wrapped in a cloud-config. Example:
+
+    #cloud-config
+    autoinstall:
+        <user data here>
+
+    To validate the user data directly, you can pass --no-expect-cloudconfig
+    """
+    )
+
+    parser = argparse.ArgumentParser(
+        prog="validate-autoinstall-user-data",
+        description=description,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+    parser.add_argument(
+        "--json-schema",
+        help="Path to the JSON schema",
+        type=argparse.FileType("r"),
+        default="autoinstall-schema.json",
+    )
+    parser.add_argument(
+        "input",
+        nargs="?",
+        help="Path to the user data instead of stdin",
+        type=argparse.FileType("r"),
+        default="-",
+    )
+    parser.add_argument(
+        "--no-expect-cloudconfig",
+        dest="expect_cloudconfig",
+        action="store_false",
+        help="Assume the data is not wrapped in cloud-config.",
+        default=True,
+    )
+    parser.add_argument(
+        "--check-link",
+        dest="check_link",
+        action="store_true",
+        help="Assert the documentation link is in the user data",
```

Comment:
I suggest help=argparse.SUPPRESS, as this is not something I expect users to be concerned about and may instead just be confusing.  I wouldn't want people to think that this is a requirement of their autoinstall data.  Please replace with a comment explaining that we just want the link in autoinstall-user-data.

Reply:
Good point, done. Thanks!